src/Components/Modal/Modal.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from "react";
+import { observer } from "mobx-react";
+import { Button, Container, Col, Row } from "react-bootstrap";
+import { ErrorToast } from "../ErrorToast/ErrorToast";
+import type { MediaPresenter } from "../MediaPresenter";
+import styles from "./Modal.module.css";
+import classnames from "classnames";
+type Props = {
+  mediaPresenter: MediaPresenter;
+};
+
+export const Modal: React.FC<Props> = observer(({ mediaPresenter }) => {
+  const [modalOpen, setModalOpen] = useState<boolean>(false);
+  const openModal = (): void => {
+    setModalOpen(true);
+  };
+  const closeModal = (): void => {
+    var modal: HTMLElement | null = document.getElementById("magicModal");
+    if (modal != null) {
+      // modal.style.display = "none";
+      setModalOpen(false);
+    }
+  };
+  const windowCloseModal = (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>
+  ): void => {
+    var modal: HTMLElement | null = document.getElementById("magicModal");
+    if (event.target === modal && modal != null) {
+      // modal.style.display = "none";
+      setModalOpen(false);
+    }
+  };
+  return (
+    <div>
+      <Button
+        variant="info"
+        onClick={openModal}
+        className={styles.createDesignBtn}
+      >
+        Create a design
+      </Button>
```

Comment:
This should live outside the Modal component, because it's not part of the modal - which means that your state should also live outside the component and then be passed into it via a prop